### ISSUE TITLE
ILC ridge scaled by the arithmetic mean of the covariance diagonal breaks wide-frequency-coverage cleaning (NILC/cMILC/GNILC)

### ISSUE THREAD

**Issue by jrcheshire** [jrcheshire/cmb-augr#50]
The ILC covariance regularization is scaled by an **arithmetic mean of the covariance diagonal**. In CMB thermodynamic units that diagonal spans many orders of magnitude across a wide frequency range, so the mean is set almost entirely by the submm channels — and the resulting ridge swamps the low-frequency channels that actually carry the CMB. Adding channels that carry ~no statistical weight then makes the cleaning dramatically **worse**, which is impossible for a constrained minimum-variance estimator and is a clear signature of a numerical bug.

## Where

`augr/nilc.py:246` — shared helper:
```python
def _ridge(cov, ridge):
    scale = jnp.trace(cov, axis1=-2, axis2=-1) / n_band   # arithmetic mean of diag
    return cov + ridge * scale[..., None, None] * eye
```

`augr/nilc.py:255` `_ilc_weights_masked` — same idea over the active mask:
```python
d_a = jnp.sum(m * diag_c, axis=-1, keepdims=True) / n_active   # arithmetic mean
ridge_diag = (ridge * d_a * m)[..., None] * eye
```

Both are reached by all three cleaners: NILC (`_global_weights`/`_localized_weights` -> `_ilc_weights_masked`), **cMILC** (`cmilc.py:230,273` via `_ridge`), **GNILC** (`gnilc.py:176-177` via `_ridge`).

## Evidence

24 bands, 10-1000 GHz, d1s1, nside=128, lmax=192, 10 sims, blind NILC. Metric is the residual expressed as sqrt(mean C_ell/(2*4pi f_sky)) over 30<=l<=180, in nK. "penalty" is post-NILC noise over the raw inverse-variance noise.

| ridge | post-NILC noise | FG residual | total | penalty |
|---|---|---|---|---|
| 0 | 0.353 | 0.210 | **0.411** | 2.00x |
| 1e-16 | 0.353 | 0.210 | **0.411** | 2.00x |
| 1e-14 | 0.352 | 0.212 | **0.411** | 1.99x |
| 1e-12 | 0.374 | 0.304 | **0.482** | 2.12x |
| **1e-10 (default)** | 0.728 | 1.605 | **1.762** | 4.12x |

The default costs a factor **4.3 in total residual and 7.6 in the foreground residual**, from regularization alone.

The channel-count ordering is the smoking gun:

| variant | ridge | total |
|---|---|---|
| 24 channels | 1e-10 | 1.762 |
| 18 channels (drop nu > 400 GHz) | 1e-10 | 0.483 |
| 24 channels | 0 | **0.411** |

At the default ridge, dropping six channels *improves* the answer 3.6x. With the ridge removed, 24 channels correctly beats 18. Adding channels can only help a constrained MV estimator in the ensemble limit, so the default-ridge ordering is unphysical.

## Mechanism

Noise-only diagonal for this band set (uK^2 sr):

| | diag spread | d_a = mean(diag) | ridge = 1e-10*d_a | as % of the 99 GHz diagonal |
|---|---|---|---|---|
| 24 ch | 6.5e9 | 1.27e3 | 1.27e-7 | **2.4%** |
| 18 ch | 2.3e2 | 1.16e-4 | 1.16e-14 | 2e-7 % |

Six channels carrying ~1e-10 of the inverse-variance weight inflate the ridge applied to *every* channel by seven orders of magnitude, because the mean is dominated by the 1000 GHz entry (3.0e4 vs 5.3e-6 at 99 GHz).

Consistency check that pins it to the foreground term: with `fg_model=None`, 24-channel and 18-channel results are **identical** (0.217 nK, 1.23x penalty). The noise diagonal is unchanged there, so a ~2% ridge alone is harmless; the damage appears only once the dust variance at submm frequencies (enormous in CMB units) dominates `d_a`, at which point the ridge added to the CMB-carrying channels exceeds their own variance.

## Suggested fix

Prewhiten before the solve — the same treatment the Fisher path got in #5/#6, where it dropped the condition number by 10+ orders of magnitude:

1. Form `D = sqrt(diag(C))`, solve the scale-free `C~ = C / (D D^T)` (unit diagonal, condition number reflecting correlation structure rather than unit dynamic range), then undo. ILC weights are invariant under this rescaling analytically, so it changes only the numerics.
2. Scale the ridge to something robust to dynamic range — each channel's own diagonal, or a geometric mean — never an arithmetic mean across many decades.
3. Regression test: assert that adding a high-noise channel never degrades the ILC residual (monotonicity in channel count), which is the invariant violated here and is cheap to check.

## Notes

- `nilc_spectra`'s `transfer` diagnostic cannot catch this: it projects a CMB-only map through the weights, and ILC weights satisfy `w.a = 1` by construction, so it reads 1.0000 regardless. It was 1.0000 throughout every run above.
- Existing presets (LiteBIRD/PICO-like, ~20-40 to ~400 GHz) have a far narrower diagonal spread, so published numbers are much less affected — but the size of the effect scales with frequency coverage, so anything reaching into the submm is exposed.
- Found while running a JPL probe-study detector-count sweep over a 10-1000 GHz band set; the artifact produced a spurious "foreground floor at 1.4 nK, unreachable at any detector count" that disappears once the ridge is fixed.
